### PR TITLE
Pick up new daqmx metadata

### PR DESCRIPTION
**Pull request by WayneDroid**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nidaqmx-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Pickup new DAQmx metadata, updated functions.py and enums.py.

### Why should this Pull Request be merged?

To support product category mioDAQ.

### What testing has been done?

`poetry run pytest tests -q` - pass

image

### CODE REVIEW

File: generated/nidaqmx/_grpc_interpreter.py [hunk 1]
```diff
@@ -2679,7 +2679,7 @@ def read_power_scalar_f64(self, task, timeout):
         return response.voltage, response.current
 
     def read_raw(self, task, num_samps_per_chan, timeout, read_array):
-        _validate_array_dtype(read_array, numpy.generic)
+        _validate_array_dtype(read_array, numpy.uint8)
```
Comment (zhindes) [inline generated/nidaqmx/_grpc_interpreter.py:2682]: Does this seem reasonable?

File: generated/nidaqmx/constants.py [hunk 2]
```diff
@@ -517,6 +517,7 @@ class ProductCategory(Enum):
     FIELD_DAQ = 16151  #: FieldDAQ.
     TEST_SCALE_CHASSIS = 16180  #: TestScale chassis.
     TEST_SCALE_MODULE = 16181  #: TestScale I/O module.
+    MIODAQ = 16182  #: mioDAQ.
```
Comment (zhindes) [inline generated/nidaqmx/constants.py:520]: I think I prefer this to `MIO_DAQ`... sure

**Review by zhindes — APPROVED** [review]

File: generated/nidaqmx/_grpc_interpreter.py (hunk 1 above)
Comment (zhindes) [reply generated/nidaqmx/_grpc_interpreter.py:2682]: I think so...

Comment (zhindes) [reply generated/nidaqmx/_grpc_interpreter.py:2682]: Doesn't seem like it; tests are failing.

```
tests\component\task\test_in_stream.py:183: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
generated\nidaqmx\task\_in_stream.py:1232: in read_into
    _, samples_read, _ = self._interpreter.read_raw(
generated\nidaqmx\_grpc_interpreter.py:2682: in read_raw
    _validate_array_dtype(read_array, numpy.uint8)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

numpy_array = array([28768, 28768, 28768, 28768, 28768, 28768, 28768, 28768, 28768,
       28768, 28768, 28768, 28768, 28768, 28768, 28768, 28768, 28768,
       28768], dtype=int16)
expected_numpy_array_dtype = <class 'numpy.uint8'>

    def _validate_array_dtype(numpy_array, expected_numpy_array_dtype):
        """Raises TypeError if array type doesn't match with expected numpy.dtype"""
        if expected_numpy_array_dtype != numpy.generic and numpy_array.dtype != expected_numpy_array_dtype:
>           raise TypeError(f"array must have data type {expected_numpy_array_dtype}")
E           TypeError: array must have data type <class 'numpy.uint8'>
```

**Review by zhindes — CHANGES_REQUESTED** [review]
That gRPC Interpreter change is breaking

**Review by zhindes — APPROVED** [review]

File: generated/nidaqmx/_grpc_interpreter.py (hunk 1 above)
Comment (WayneDroid) [reply generated/nidaqmx/_grpc_interpreter.py]: The metadata is reverted, the tests passing now.

**Pull request merged**